packages/@aws-cdk/aws-codebuild/lib/project.ts

```diff
@@ -853,6 +882,24 @@ export class Project extends ProjectBase {
     }
   }
 
+  public enableBatchBuilds(): IEnableBatchBuildsReturn | undefined {
+    if (!this._batchServiceRole) {
+      this._batchServiceRole = new iam.Role(this, 'BatchServiceRole', {
+        roleName: PhysicalName.GENERATE_IF_NEEDED,
+        assumedBy: new iam.ServicePrincipal('codebuild.amazonaws.com'),
+      });
+      this._batchServiceRole.addToPrincipalPolicy(new iam.PolicyStatement({
+        resources: [Lazy.string({
+          produce: () => this.projectArn,
+        })],
+        actions: ['codebuild:StartBuild', 'codebuild:StopBuild', 'codebuild:RetryBuild'],
+      }));
```

Comment:
Don't we need some batch permissions here? Like this? https://github.com/aws/aws-cdk/pull/12181/files?file-filters%5B%5D=.ts#diff-7902b8e1bfd8571028c1c97116a58c4f6e090f92d7df3490e16e12defbfa2fab

Reply:
I don't think so. The docs don't mention it and it worked when I tested this in a different account.

Think this is so that internally non batch builds can be triggered as part of a batch. Don't think a batch can trigger another batch?